common/dbIntrospection.ts

```diff
@@ -0,0 +1,82 @@
+import { Client } from "pg";
+import connectDBClient, { testDBClientConnection } from "./client";
+import { TableSchema } from "./interfaces";
+
+export default async function introspectDB(client: Client) {
+  if (!client) {
+    return;
+  }
```

Comment:
caller receives undefined when the client is falsy? 
1. why would the client be falsy - it's not typed as nullish. 
2. why would you not throw an error in this case even if it was falsy?